runtime/opt/taupage/runtime/Docker.py

```diff
@@ -356,46 +376,75 @@ def wait_for_health_check(config: dict):
     sys.exit(2)
 
 
-def is_valid_source(source):
-    '''
-    >>> is_valid_source('')
-    False
+def parse_image_tag(source):
+    '''Parse a docker tag into image, org, name, tag, throwing an error if any of the components are missing.
 
-    >>> is_valid_source('foo/bar')
-    False
+    >>> parse_image_tag('')
+    Traceback (most recent call last):
+    ...
+    ValueError: Image tag not specified
 
-    >>> is_valid_source('foo/bar:latest')
-    False
+    >>> parse_image_tag('nginx:1.2.3')
+    Traceback (most recent call last):
+    ...
+    ValueError: No registry specified or invalid image name: nginx
 
-    >>> is_valid_source('foo/bar:1.0-SNAPSHOT')
-    False
+    >>> parse_image_tag('foo/nginx:1.2.3')
+    Traceback (most recent call last):
+    ...
+    ValueError: No registry specified or invalid image name: foo/nginx
 
-    >>> is_valid_source('foo/bar:1.0')
-    True
-    '''
-    parts = source.rsplit(':', 1)
-    if len(parts) != 2:
-        # missing tag
-        return False
-    _, tag = parts
-    if tag == 'latest':
-        # mutable "latest" images are not allowed
-        return False
-    elif 'SNAPSHOT' in tag:
-        # mutable *-SNAPSHOT images are not allowed
-        return False
-    return True
+    >>> parse_image_tag('registry.example.org/foo/nginx')
+    Traceback (most recent call last):
+    ...
+    ValueError: Image tag not specified
 
+    >>> parse_image_tag('registry.example.org/foo/nginx:')
+    Traceback (most recent call last):
+    ...
+    ValueError: Image tag not specified
 
-def main(args):
+    >>> parse_image_tag('registry.example.org/foo/nginx:latest')
```

Comment:
Parsing a string into an image dict seems to also exist in the PierOne API client code: https://github.com/zalando-stups/pierone-cli/blob/master/pierone/api.py#L27